Most of the Python version is complete

ebodine-jump opened the pull request:
The python version should be functional, please feel free to test it out and give some feedback in this PR.

The README is not complete.

kyle-randolph commented:
Pretty strong work so far here, I did some testing and it looks like the script is running the authorization component pretty well. After reviewing the code, I did have a few questions: 

What is the purpose of 'binddn_prefix' and 'binddn_suffix' outlined in the config.json file? I don't see either of these values used in the python_ldap_authentication.py script. Also, we may want to omit your organization information for the example variables on lines 153-157 or populate it with dummy info. 

In addition, it looks like this script is currently going through authorization of the user existing in the proper group; however, I did not see user authentication occurring in
this script as only the ldap binding user password (password_field) is being requested. 

On the last piece, can you include more specific error/success messaging? Ideally, it would exit on one of the try...catch blocks when the specified org_id is invalid. In addition, I'm not sure if this is the most common means of capturing success with Python but would it make sense to just provide a success code of 200 on this so that admins can then take that 200 success to allow user authentication? Or they can populate whatever other error message is raised for a dialogue box in their application.

ebodine-jump commented:
Inline

On Tue, Nov 21, 2017 at 12:40 PM, kyle-randolph <notifications@github.com>
wrote:

> Pretty strong work so far here, I did some testing and it looks like the
> script is running the authorization component pretty well. After reviewing
> the code, I did have a few questions:
>
> What is the purpose of 'binddn_prefix' and 'binddn_suffix' outlined in the
> config.json file? I don't see either of these values used in the
> python_ldap_authentication.py script. Also, we may want to omit your
> organization information for the example variables on lines 153-157 or
> populate it with dummy info.
>
removed binddn_prefix & binddn_suffix.  I will remove the Org info when we
think the python script is where we want it; filling out the getpass()
questions over and over while testing is a big time sink.

> In addition, it looks like this script is currently going through
> authorization of the user existing in the proper group; however, I did not
> see user authentication occurring in
> this script as only the ldap binding user password (password_field) is
> being requested.
>
Not sure I understand.

> On the last piece, can you include more specific error/success messaging?
> Ideally, it would exit on one of the try...catch blocks when the specified
> org_id is invalid. In addition, I'm not sure if this is the most common
> means of capturing success with Python but would it make sense to just
> provide a success code of 200 on this so that admins can then take that 200
> success to allow user authentication? Or they can populate whatever other
> error message is raised for a dialogue box in their application.
>
I added "CRITICAL" to failure messages, added a "Success!" message at the
end (having scripts that run successfully exit with something other than 0
is really, really bad form) - that is probably different for Javascript (i
have to do some homework there).  Invalid Org Id is caught, I changed the
exception message to reflect it.

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/TheJumpCloud/support/pull/6#issuecomment-346138301>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AZKqRa17o5kXF7ddJrkfMnIhb6b_6mBFks5s4ycSgaJpZM4QmQgi>
> .
>



-- 
Erick Bodine

ebodine-jump commented:
Kyle:  added the ruby version and comments + user for the python version.  let me know what you think

jc-rob commented:
it looks like these are all ldap examples?  If that's the case, we should throw all these into /scripts/authentication_examples/ldap to account for any future expansion of examples against a different endpoint, e.g. radius, saml, oauth(someday), etc...

ebodine-jump commented:
Yes, these are all ldap examples.  Sure, I am good with doing that.

On Wed, Nov 29, 2017 at 5:06 PM, jc-rob <notifications@github.com> wrote:

> it looks like these are all ldap examples? If that's the case, we should
> throw all these into /scripts/authentication_examples/ldap to account for
> any future expansion of examples against a different endpoint, e.g. radius,
> saml, oauth(someday), etc...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/TheJumpCloud/support/pull/6#issuecomment-348038802>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AZKqRcKh0yDQQ2p8yKJ7jpqCYh6lAfBgks5s7fF2gaJpZM4QmQgi>
> .
>



-- 
Erick Bodine

kyle-randolph commented:
Awesome work on these Erick!

I did find a few things that will need to be addressed on the ruby and python scripts. 

Ruby:
- Currently does not filter against the Groups section. 
- 2nd Bind Authentication does not output an error code when username or password is invalid

Python:
- It looks like your filler org-id, bind-user, etc are currently commented out while the input prompts were commented 
- Provide accurate error messaging when an incorrect password is entered for the system user

kyle-randolph commented:
Reviewed the recent changes and it looks like the Python script is still allowing the system user to complete the LDAP auth with an incorrect password. I tested the Groups search functionality on Ruby and unfortunately it looks like that is not working currently.

ebodine-jump commented:
@kyle-randolph   merged and good to go
